File: js/anthropic/examples/anthropic-wrapped.ts
/**
 * Example: wrap an Anthropic client with AgentLoop in 3 lines.
 *
 * Run:
 *   npm install @anthropic-ai/sdk @agentloop-sdk/core @agentloop-sdk/anthropic
 *   export ANTHROPIC_API_KEY=sk-ant-...
 *   export AGENTLOOP_API_KEY=ak_...
 *   tsx examples/anthropic-wrapped.ts "what's the pix limit at night?"
 */

import Anthropic from "@anthropic-ai/sdk";
import { AgentLoop } from "@agentloop-sdk/core";
import { wrapAnthropic } from "@agentloop-sdk/anthropic";

const loop = new AgentLoop({
  apiKey: process.env.AGENTLOOP_API_KEY ?? (() => { throw new Error("AGENTLOOP_API_KEY is required"); })(),
});

// Wrap once at startup. Every messages.create call on the returned
// client now fires AgentLoop hooks automatically.
const anthropic = wrapAnthropic(new Anthropic(), {
  loop,

  // Optional: detect quality signals from the assistant's response.
  detectSignals: (_question, answer, memories) => {
    const signals: Record<string, boolean> = {};
    if (/i['’]?m not sure|i don['’]?t know|contact support/i.test(answer)) {
      signals.agent_punted = true;
    }
    if (/\$|R\$|%/.test(answer)) signals.factual_claim = true;
    if (signals.factual_claim && memories.length === 0) {
      signals.low_confidence = true;
    }
    return Object.keys(signals).length > 0 ? signals : undefined;
  },
});

export async function ask(question: string, userId: string = "demo_user"): Promise<string> {
  const msg = await anthropic.messages.create({
    model: "claude-opus-4-7",
    max_tokens: 1024,
    system: "You are a helpful customer support assistant. Be concise.",
    messages: [{ role: "user", content: question }],
    agentloop: { userId },
  });

  // Anthropic's content is an array of blocks. Extract text.
  return msg.content
    .filter((c): c is Anthropic.TextBlock => c.type === "text")
    .map((c) => c.text)
    .join("");
}

// CLI harness
if (import.meta.url === `file://${process.argv[1]}`) {
  const q = process.argv[2] ?? "What's the Pix limit at night?";
  ask(q)
    .then((a) => console.log("\nAnswer:", a))
    .catch((err) => { console.error(err); process.exit(1); });
}

File: round14c-delivery/py/anthropic/examples/anthropic_wrapped.py
"""
Example: wrap an Anthropic client with AgentLoop. Zero setup.

Run:
    pip install agentloop-sdk agentloop-anthropic anthropic
    export ANTHROPIC_API_KEY=sk-ant-...
    export AGENTLOOP_API_KEY=ak_...
    python examples/anthropic_wrapped.py "what's the pix limit at night?"
"""

from __future__ import annotations

import os
import sys

from anthropic import Anthropic

from agentloop import AgentLoop
from agentloop_anthropic import wrap_anthropic

AGENTLOOP_KEY = os.environ.get("AGENTLOOP_API_KEY")
ANTHROPIC_KEY = os.environ.get("ANTHROPIC_API_KEY")
if not AGENTLOOP_KEY or not ANTHROPIC_KEY:
    raise SystemExit(
        "Set AGENTLOOP_API_KEY and ANTHROPIC_API_KEY env vars before running."
    )


def detect_signals(_q: str, answer: str, memories: list) -> dict | None:
    signals: dict[str, bool] = {}
    low = answer.lower()
    if "not sure" in low or "contact support" in low:
        signals["agent_punted"] = True
    if any(c in answer for c in "$R%"):
        signals["factual_claim"] = True
    if signals.get("factual_claim") and not memories:
        signals["low_confidence"] = True
    return signals or None


anthropic = wrap_anthropic(
    Anthropic(api_key=ANTHROPIC_KEY),
    loop=AgentLoop(api_key=AGENTLOOP_KEY),
    detect_signals=detect_signals,
)


def ask(question: str, user_id: str = "demo_user") -> str:
    msg = anthropic.messages.create(
        model="claude-opus-4-7",
        max_tokens=1024,
        system="You are a helpful customer support assistant. Be concise.",
        messages=[{"role": "user", "content": question}],
        agentloop={"user_id": user_id},
    )
    # Anthropic's content is a list of blocks — extract text
    return "".join(
        block.text for block in msg.content if getattr(block, "type", None) == "text"
    )


if __name__ == "__main__":
    q = sys.argv[1] if len(sys.argv) > 1 else "What's the Pix limit at night?"
    print(ask(q))
